codelab/02_kpt-deploy/tutorial.md

```diff
@@ -0,0 +1,240 @@
+Or just follow this codelab, we will explain what happens in each step.
+
+#### Install `skaffold`
+
+*   Check if `skaffold` is installed and the installed version is >= v1.17.0
+    ```bash
+    skaffold version
+    ```
+
+*   If you haven't installed `skaffold` previously, run
+    ```bash
+    curl -Lo skaffold https://storage.googleapis.com/skaffold/builds/latest/skaffold-linux-amd64 && \
+    sudo install skaffold /usr/local/bin/ | bash
+    ```  
+*   To upgrade skaffold to a newer version, run
+    ```bash
+    sudo rm -f /usr/local/bin/skaffold && curl -Lo skaffold https://storage.googleapis.com/skaffold/builds/latest/skaffold-linux-amd64 && \
+    sudo install skaffold /usr/local/bin/ | bash
+    ```
+    
+#### Install `kpt`
+
+*   Check if `kpt` is installed and the installed version is >= 0.34.0
+    ```bash
+    kpt version
+    ```
+*   If you haven't installed `kpt` previously, run
+    ```bash
+    sudo apt-get install google-cloud-sdk-kpt
+    ```  
+    
+#### Install `kustomize`
+
+*   Check if `kustomize` is installed and the installed version is >= v3.4.3
+    ```bash
+    kustomize version
+    ```
+*   If you haven't installed `kustomize` previously, run
+    ```bash
+    curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash &&  sudo mv kustomize /usr/local/go/bin
+    ```  
+    
+## Getting started
+
+Time to complete: **About 3 minutes**
+
+
+*   Run this command to start your local minikube cluster:
+    
+    ```bash
+    minikube start
+    ```
+    
+*   Once your cluster is set up, the following message is displayed:
+    
+    ```terminal
+    Done! kubectl is now configured to use "minikube"
+    ```
+    
+*   Now let's use `kpt pkg` to download the application example. 
+        
+    ```bash
+    kpt pkg get https://github.com/GoogleContainerTools/skaffold.git/codelab/02_kpt-deploy/resources/sample-app guestbook-cl2 && cd guestbook-cl2
+    ```
+
+<walkthrough-pin-section-icon></walkthrough-pin-section-icon>
+**Tips**
+>  If you haven't used `skaffold dev`, you can go through this [codelab](https://github.com/GoogleContainerTools/skaffold/tree/master/codelab/01_kpt-validate).
+
+## Deploy
+Time to complete: **About 2 minutes**
+
+`Skaffold` supports using `kpt` to deploy your configurations to the cluster.
+
+This deployment is equivalent to [`kpt live`](https://googlecontainertools.github.io/kpt/reference/live/) which reconciles the resources in their live states. 
+
+Compared to other deployment tools, one key advantage of the kpt deployment is the "pruning" feature. "Prune" means removing the resources from the cluster if the resources do not appear in the applied configuration. 
+
+#### "prune"
+
+You may have used the following command to remove resources that are not shown up in the `DIR`. 
+
+```terminal
+kubectl apply --prune -f DIR
+``` 
+
+**Note** The resources may not always be pruned even though they are not in the DIR. This  
```

Comment:
Is there something missing at the end here?

Reply:
done.